File: Basics/Basics/ViewController.m
//
//  ViewController.m
//  Basics
//
//  Created by Ebony Nyenya on 1/5/15.
//  Copyright (c) 2015 Ebony Nyenya. All rights reserved.
//

#import "ViewController.h"

//  Create 2 View Controllers on the main storyboard

// View Controller 1
// - 2 buttons (one will clear textfields, one will log all textfields)
// - 3 textfields (username, email, password)
// - password field should look like one by hiding the text with dots (look in the right panel)
// - email field should use email keyboard (also in right panel)

// View Controller 2
// - 4 buttons with different color backgrounds (one method for all 4 buttons)
// - each button changes the main view controller's background color

// Add constraints
// Make the buttons on view controller 2 circles with no text

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)username:(UITextField *)sender {
}
- (IBAction)clear:(UIButton *)sender {
    
    self.username.text = @"";
    self.email.text = @"";
    self.password.text = @"";

}



- (IBAction)log:(UIButton *)sender {
    
    NSString * username = self.username.text;
    
    NSLog(@"Username is %@", username);
    
    
    NSString * email = self.email.text;
    
    NSLog(@"Email is %@", email);
    
    
    NSString * password = self.password.text;
    
    NSLog(@"Username is %@", password);

    
}



@end

File: LoginParse/LoginParse/ViewController.swift
//
//  ViewController.swift
//  LoginParse
//
//  Created by Ebony Nyenya on 1/27/15.
//  Copyright (c) 2015 Ebony Nyenya. All rights reserved.
//
//HW 1/26/15

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var usernameField: UITextField!
    
    
    @IBOutlet weak var emailField: UITextField!
    
    @IBOutlet weak var passwordField: UITextField!
    
    
    @IBAction func loginUser(sender: AnyObject) {
        
        var fieldValues: [String] = [usernameField.text, emailField.text, passwordField.text]
        
        if find(fieldValues, "") != nil {
            
            //all fields are not filled in
            var alertViewController = UIAlertController(title: "Submission Error", message: "All fields need to be filled in", preferredStyle: UIAlertControllerStyle.Alert)
            
            var defaultAction = UIAlertAction(title: "OK", style: .Default, handler: nil)
            
            alertViewController.addAction(defaultAction)
            
            presentViewController(alertViewController, animated: true, completion: nil)
        } else {
            
            //all fields are filled in
            println("All fields are filled in and login complete")
            
            var userQuery = PFUser.query()
            userQuery.whereKey("username", equalTo: usernameField.text)
            
            userQuery.findObjectsInBackgroundWithBlock({ (objects, error) -> Void in
                
                if objects.count > 0 {
                    
                    self.login()
                    
                } else {
                    
                    self.signUp()
                }
            })
            
        }
    }
    
    func login() {
        
        PFUser.logInWithUsernameInBackground(usernameField.text, password:passwordField.text) {
            (user: PFUser!, error: NSError!) -> Void in
            
            if user != nil {
                
                // Do stuff after successful login.
                println("logged in as \(user)")
                
            } else {
                // The login failed. Check error to see why.
            }
            
            
        }
        
    }
    
    func signUp() {
        
        var user = PFUser()
        user.username = usernameField.text
        user.password = passwordField.text
        user.email = emailField.text
        // other fields can be set just like with PFObject
        //user["phone"] = "415-392-0202"
        
        user.signUpInBackgroundWithBlock {
            (succeeded: Bool!, error: NSError!) -> Void in
            if error == nil {
                // Hooray! Let them use the app now.
                
              println(user)
                
              self.usernameField.text = ""
              self.passwordField.text = ""
              self.emailField.text = ""
                
                
            } else {
                let errorString = error.userInfo?["error"] as NSString
                // Show the errorString somewhere and let the user try again.
            }
        }
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

